Sources/Apodini/Request/PathParameter.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+
+/// A `@PathComponent` can be used in `Component`s to indicate that a part of a path is a parameter and can be read out in a `Handler`
+@propertyWrapper
+public struct PathParameter<Element> {
+    var id = UUID()
+    
+    /// You can never access the wrapped value of a @PathParameter.
+    /// Please use a `@Parameter` wrapped property within a `Handler` to access the path property.
+    public var wrappedValue: Element {
+        fatalError("""
+            You can never access the wrapped value of a @PathParameter.
+            Please use a `@Parameter` wrapped property within a `Handler` to access the path property.
+        """)
+    }
+    
+    /// Accessing the projected value allows you to pass the `@PathParameter` to a `Handler` or `Component`
+    public var projectedValue: PathParameter {
+        self
+    }
```

Comment:
Doesn't the projected value of a PathParameter need to return an instance of `@Parameter`?
Judging by the initializer inside the `@Parameter` property wrapper I think `self` should be replaced with `Parameter(self)`? Not sure what the intention here is.

```swift
// ...
struct ExampleNestComponent: Handler {
    @Parameter var birdID: Bird.ID
    @Parameter var nestName: String?

    // ...
}

struct TestWebService: WebService, Component {
    @PathParameter var birdID: Bird.ID

    var content: some Component {
        Group("api", "birds", birdID) {
            ExampleBirdComponent( /* >>> */ birdID: $birdID  /* <<< */) // PathParameter doesn't match type Parameter
            Group("nests") {
                ExampleNestComponent( /* >>> */ birdID: $birdID  /* <<< */)
            }
        }
    }
}
// ...
```

Reply:
The current implementation has an initializer for `@Parameter` that gets a `@PathParameter` as an input. Therefore you can not pass `@Parameters` down a component tree but only `@Parameters` can be easily passed down the component tree.